File: generator.rb
#!/usr/bin/env ruby
#
require 'json'

begin
	file = File.open(ARGV[0], "r")
	results = `wc -l #{ARGV[0]}`.strip.split(' ')[0]
	puts "<html><head><meta charset=\"utf-8\" />\n<title>Table of items</title></head>"
	puts "#{results} results generated at #{File.ctime(ARGV[0]).to_s}<br>"
	puts "<table border='1'>"
	puts "<tr>"
	puts "<th>帖子</th>"
	puts "<th>作者</th>"
	puts "<th>贴吧</th>"
	puts "<th>预览</th>"
	puts "<th>日期</th>"
	puts "<th>地址</th>"
	puts "</tr>"
	file.each do |line|
		puts "<tr>"
		json = JSON.parse(line)
		print "<td>" + json['title'] + "</td>\n"
		print "<td>" + json['author'] + "</td>\n"
		print "<td>" + json['tieba'] + "</td>\n"
		print "<td>" + json['preview'] + "</td>\n"
		print "<td>" + json['date'] + "</td>\n"
		print "<td><a href=\"" + json['url'] + "\">link</a></td>\n"
		puts "</tr>"
	end
	puts "</table></html>"
	file.close if not file.nil?
end

File: generator.pl
#!/usr/bin/env perl
#
use JSON;
use Cwd;
use utf8;
use File::stat;
use Time::localtime;
<<<<<<< HEAD
=======
use File::Basename;
use Try::Tiny;
>>>>>>> 080e79c... update generator.pl to skip broken json

use open ':std', ':encoding(UTF-8)';

open($fh, '<:encoding(UTF-8)', $ARGV[0]) or die "open file failed.";

print("<html>\n");
print("<head>\n");
print("<meta charset=\"utf-8\" />\n");
print("<title>Table of items</title>\n");
print("</head><body>\n");
my $wc = `wc -l < $ARGV[0]`;
chomp;
$wc =~ s/\s//g;
print("$wc results generated at ". ctime(stat($fh)->ctime) . "<br>\n");
print("<table border='1'>\n");
print("<tr><th>标题</th><th>作者</th><th>贴吧</th><th>预览</th><th>日期</th><th>地址</th></tr>\n");

while(<$fh>){
	try{
		chomp;
		$json = JSON->new->utf8->decode($_);
		print("<tr>\n");
		print("<td>". $json->{'title'}.	"</td>\n");
		print("<td>". $json->{'author'}.	"</td>\n");
		print("<td>". $json->{'tieba'}.	"</td>\n");
		print("<td>". $json->{'preview'}.	"</td>\n");
		print("<td>", $json->{'date'}.	"</td>\n");
		print("<td><a href=\"", $json->{'url'}.	"\" >link</a></td>\n");
		print("</tr>\n");
	}catch{
		next;
	}
}
close($fh);
print("</table></body></html>\n");
